datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/snmp/validate_snmp_profiles.py

```diff
@@ -1,21 +1,75 @@
 import click
+import glob
+import os
 
 from .....fs import dir_exists, file_exists
-from ...console import CONTEXT_SETTINGS, abort, echo_failure
+from ...console import CONTEXT_SETTINGS, abort, echo_failure, echo_success, echo_warning, echo_info
 from . import validators
 
+from .validators.utils import (
+    initialize_path,
+    exist_profile_in_path
+)
+
+MESSAGE_METHODS = {'success': echo_success, 'warning': echo_warning, 'failure': echo_failure, 'info': echo_info}
 
 @click.command("validate-profile", short_help="Validate SNMP profiles", context_settings=CONTEXT_SETTINGS)
 @click.option('-f', '--file', help="Path to a profile file to validate")
 @click.option('-d', '--directory', help="Path to a directory of profiles to validate")
 @click.option('-v', '--verbose', help="Increase verbosity of error messages", is_flag=True)
 def validate_profile(file, directory, verbose):
```

Comment:
I'm getting some errors with `integrations-core` snmp profiles:

```
 ddev meta snmp validate-profile -d snmp/datadog_checks/snmp/data/profiles
...
Start validation of profile snmp/datadog_checks/snmp/data/profiles/cisco-csr1000v.yaml:
FAILED
Schema successfully validated
metric with OID 1.3.6.1.2.1.4.31.1.1.4 is duplicated in profiles:
|------> _generic-ip.yaml:10
|------> _generic-ip.yaml:10

metric with OID 1.3.6.1.2.1.4.31.1.1.6 is duplicated in profiles:
|------> _generic-ip.yaml:12
|------> _generic-ip.yaml:12

metric with OID 1.3.6.1.2.1.4.31.1.1.7 is duplicated in profiles:
|------> _generic-ip.yaml:14
|------> _generic-ip.yaml:14

metric with OID 1.3.6.1.2.1.4.31.1.1.8 is duplicated in profiles:
|------> _generic-ip.yaml:16
|------> _generic-ip.yaml:16

metric with OID 1.3.6.1.2.1.4.31.1.1.9 is duplicated in profiles:
|------> _generic-ip.yaml:18
|------> _generic-ip.yaml:18

metric with OID 1.3.6.1.2.1.4.31.1.1.10 is duplicated in profiles:
|------> _generic-ip.yaml:20
|------> _generic-ip.yaml:20

metric with OID 1.3.6.1.2.1.4.31.1.1.11 is duplicated in profiles:
|------> _generic-ip.yaml:22
|------> _generic-ip.yaml:22

metric with OID 1.3.6.1.2.1.4.31.1.1.13 is duplicated in profiles:
|------> _generic-ip.yaml:24
|------> _generic-ip.yaml:24

metric with OID 1.3.6.1.2.1.4.31.1.1.14 is duplicated in profiles:
|------> _generic-ip.yaml:26
|------> _generic-ip.yaml:26

metric with OID 1.3.6.1.2.1.4.31.1.1.15 is duplicated in profiles:
|------> _generic-ip.yaml:28
|------> _generic-ip.yaml:28

metric with OID 1.3.6.1.2.1.4.31.1.1.16 is duplicated in profiles:
|------> _generic-ip.yaml:30
|------> _generic-ip.yaml:30

metric with OID 1.3.6.1.2.1.4.31.1.1.17 is duplicated in profiles:
|------> _generic-ip.yaml:32
|------> _generic-ip.yaml:32

metric with OID 1.3.6.1.2.1.4.31.1.1.19 is duplicated in profiles:
|------> _generic-ip.yaml:34
|------> _generic-ip.yaml:34

metric with OID 1.3.6.1.2.1.4.31.1.1.21 is duplicated in profiles:
|------> _generic-ip.yaml:36
|------> _generic-ip.yaml:36

metric with OID 1.3.6.1.2.1.4.31.1.1.22 is duplicated in profiles:
|------> _generic-ip.yaml:38
|------> _generic-ip.yaml:38

metric with OID 1.3.6.1.2.1.4.31.1.1.24 is duplicated in profiles:
|------> _generic-ip.yaml:40
|------> _generic-ip.yaml:40

metric with OID 1.3.6.1.2.1.4.31.1.1.25 is duplicated in profiles:
|------> _generic-ip.yaml:42
|------> _generic-ip.yaml:42

metric with OID 1.3.6.1.2.1.4.31.1.1.26 is duplicated in profiles:
|------> _generic-ip.yaml:44
|------> _generic-ip.yaml:44

metric with OID 1.3.6.1.2.1.4.31.1.1.27 is duplicated in profiles:
|------> _generic-ip.yaml:46
|------> _generic-ip.yaml:46

metric with OID 1.3.6.1.2.1.4.31.1.1.28 is duplicated in profiles:
|------> _generic-ip.yaml:48
|------> _generic-ip.yaml:48

metric with OID 1.3.6.1.2.1.4.31.1.1.29 is duplicated in profiles:
|------> _generic-ip.yaml:50
|------> _generic-ip.yaml:50

metric with OID 1.3.6.1.2.1.4.31.1.1.31 is duplicated in profiles:
|------> _generic-ip.yaml:52
|------> _generic-ip.yaml:52

metric with OID 1.3.6.1.2.1.4.31.1.1.33 is duplicated in profiles:
|------> _generic-ip.yaml:54
|------> _generic-ip.yaml:54

metric with OID 1.3.6.1.2.1.4.31.1.1.35 is duplicated in profiles:
|------> _generic-ip.yaml:56
|------> _generic-ip.yaml:56

metric with OID 1.3.6.1.2.1.4.31.1.1.37 is duplicated in profiles:
|------> _generic-ip.yaml:58
|------> _generic-ip.yaml:58

metric with OID 1.3.6.1.2.1.4.31.1.1.39 is duplicated in profiles:
|------> _generic-ip.yaml:60
|------> _generic-ip.yaml:60

metric with OID 1.3.6.1.2.1.4.31.1.1.41 is duplicated in profiles:
|------> _generic-ip.yaml:62
|------> _generic-ip.yaml:62

metric with OID 1.3.6.1.2.1.4.31.1.1.43 is duplicated in profiles:
|------> _generic-ip.yaml:64
|------> _generic-ip.yaml:64

metric with OID 1.3.6.1.2.1.4.31.1.1.45 is duplicated in profiles:
|------> _generic-ip.yaml:66
|------> _generic-ip.yaml:66

metric with OID 1.3.6.1.2.1.4.31.3.1.6 is duplicated in profiles:
|------> _generic-ip.yaml:84
|------> _generic-ip.yaml:84

metric with OID 1.3.6.1.2.1.4.31.3.1.7 is duplicated in profiles:
|------> _generic-ip.yaml:86
|------> _generic-ip.yaml:86

metric with OID 1.3.6.1.2.1.4.31.3.1.8 is duplicated in profiles:
|------> _generic-ip.yaml:88
|------> _generic-ip.yaml:88

metric with OID 1.3.6.1.2.1.4.31.3.1.9 is duplicated in profiles:
|------> _generic-ip.yaml:90
|------> _generic-ip.yaml:90

metric with OID 1.3.6.1.2.1.4.31.3.1.10 is duplicated in profiles:
|------> _generic-ip.yaml:92
|------> _generic-ip.yaml:92

metric with OID 1.3.6.1.2.1.4.31.3.1.11 is duplicated in profiles:
|------> _generic-ip.yaml:94
|------> _generic-ip.yaml:94

metric with OID 1.3.6.1.2.1.4.31.3.1.13 is duplicated in profiles:
|------> _generic-ip.yaml:96
|------> _generic-ip.yaml:96

metric with OID 1.3.6.1.2.1.4.31.3.1.14 is duplicated in profiles:
|------> _generic-ip.yaml:98
|------> _generic-ip.yaml:98

metric with OID 1.3.6.1.2.1.4.31.3.1.15 is duplicated in profiles:
|------> _generic-ip.yaml:100
|------> _generic-ip.yaml:100

metric with OID 1.3.6.1.2.1.4.31.3.1.16 is duplicated in profiles:
|------> _generic-ip.yaml:102
|------> _generic-ip.yaml:102

metric with OID 1.3.6.1.2.1.4.31.3.1.17 is duplicated in profiles:
|------> _generic-ip.yaml:104
|------> _generic-ip.yaml:104

metric with OID 1.3.6.1.2.1.4.31.3.1.19 is duplicated in profiles:
|------> _generic-ip.yaml:106
|------> _generic-ip.yaml:106

metric with OID 1.3.6.1.2.1.4.31.3.1.21 is duplicated in profiles:
|------> _generic-ip.yaml:108
|------> _generic-ip.yaml:108

metric with OID 1.3.6.1.2.1.4.31.3.1.24 is duplicated in profiles:
|------> _generic-ip.yaml:110
|------> _generic-ip.yaml:110

metric with OID 1.3.6.1.2.1.4.31.3.1.25 is duplicated in profiles:
|------> _generic-ip.yaml:112
|------> _generic-ip.yaml:112

metric with OID 1.3.6.1.2.1.4.31.3.1.26 is duplicated in profiles:
|------> _generic-ip.yaml:114
|------> _generic-ip.yaml:114

metric with OID 1.3.6.1.2.1.4.31.3.1.27 is duplicated in profiles:
|------> _generic-ip.yaml:116
|------> _generic-ip.yaml:116

metric with OID 1.3.6.1.2.1.4.31.3.1.28 is duplicated in profiles:
|------> _generic-ip.yaml:118
|------> _generic-ip.yaml:118

metric with OID 1.3.6.1.2.1.4.31.3.1.29 is duplicated in profiles:
|------> _generic-ip.yaml:120
|------> _generic-ip.yaml:120

metric with OID 1.3.6.1.2.1.4.31.3.1.31 is duplicated in profiles:
|------> _generic-ip.yaml:122
|------> _generic-ip.yaml:122

metric with OID 1.3.6.1.2.1.4.31.3.1.33 is duplicated in profiles:
|------> _generic-ip.yaml:124
|------> _generic-ip.yaml:124

metric with OID 1.3.6.1.2.1.4.31.3.1.35 is duplicated in profiles:
|------> _generic-ip.yaml:126
|------> _generic-ip.yaml:126

metric with OID 1.3.6.1.2.1.4.31.3.1.37 is duplicated in profiles:
|------> _generic-ip.yaml:128
|------> _generic-ip.yaml:128

metric with OID 1.3.6.1.2.1.4.31.3.1.39 is duplicated in profiles:
|------> _generic-ip.yaml:130
|------> _generic-ip.yaml:130

metric with OID 1.3.6.1.2.1.4.31.3.1.41 is duplicated in profiles:
|------> _generic-ip.yaml:132
|------> _generic-ip.yaml:132

metric with OID 1.3.6.1.2.1.4.31.3.1.43 is duplicated in profiles:
|------> _generic-ip.yaml:134
|------> _generic-ip.yaml:134

metric with OID 1.3.6.1.2.1.4.31.3.1.45 is duplicated in profiles:
|------> _generic-ip.yaml:136
|------> _generic-ip.yaml:136

Start validation of profile snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:
FAILED
Schema successfully validated
metric with OID 1.3.6.1.4.1.318.1.1.1.11.1.1.0 is duplicated in profiles:
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:62
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:70
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:78
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:86
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:94
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:102
|------> snmp/datadog_checks/snmp/data/profiles/apc_ups.yaml:110
```

For `apc_ups.yaml` case, those are valid cases, it's the same OID, but used for different metrics.

Reply:
Yes. I haven't taken into account stream_flags.
I changed it now to consider it.